[FLINK-39392][table] Support conditional traits for PTFs (#27886)
apache/flink: docs/content/docs/sql/reference/queries/changelog.md

| Parameter    | Required | Description |
|:-------------|:---------|:------------|

> | `input`      | Yes      | The input table. With `PARTITION BY`, rows with the same key are co-located for parallel execution. Without `PARTITION BY`, each row is processed independently. Accepts insert-only, retract, and upsert tables. For upsert tables, providing `PARTITION BY` is recommended for better performance. |

Reviewer: ```suggestion
| `input`      | Yes      | The input table. With `PARTITION BY`, rows with the same key are co-located for parallel execution (set semantics). Without `PARTITION BY`, each row is processed independently (row semantics). Accepts insert-only, retract, and upsert tables. |
```

Author: updated docs

-- +I[id:2, op:'DELETE',       name:'Bob',   cnt:1]
```

#### Without PARTITION BY

> ```sql

Reviewer: Let's remove all PARTITION BY examples for now. A default TO_CHANGELOG example should always be without PARTITION BY. They do not provide any benefit but rather add exchange overhead.

Author: Done

```sql
SELECT * FROM TO_CHANGELOG(
  input => TABLE source_table [PARTITION BY key_col],
  [op => DESCRIPTOR(op_column_name),]
  [op_mapping => MAP['INSERT', 'I', 'DELETE', 'D', ...]]
)
```

### Parameters

| Parameter    | Required | Description                                                                                                                                                                                                                                                                                                                                              |
|:-------------|:---------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `input`      | Yes      | The input table. With `PARTITION BY`, rows with the same key are co-located and run in the same operator instance. Without `PARTITION BY`, each row is processed independently. Accepts insert-only, retract, and upsert tables. For upsert tables, providing `PARTITION BY` is recommended for better performance.                                      |

Reviewer: ```suggestion
| `input`      | Yes      | The input table. With `PARTITION BY`, rows with the same key are co-located and run in the same operator instance. Without `PARTITION BY`, each row is processed independently. Accepts insert-only, retract, and upsert tables. For upsert tables, a provided `PARTITION BY` must match the upsert key of the subquery.                                      |
```

Author: The key doesn't have to match exactly, it can be a subset. Updating with "For upsert tables, the provided `PARTITION BY` key should match or be a subset of the upsert key of the subquery"
